identity/src/commonMain/kotlin/com/android/identity/nfc/ResponseApdu.kt

```diff
@@ -39,11 +42,10 @@ data class ResponseApdu(
      * @return the bytes of the APDU.
      */
     fun encode(): ByteArray {
-        val bsb = ByteStringBuilder()
-        bsb.append(payload)
-        bsb.append(sw1.toByte())
-        bsb.append(sw2.toByte())
-        return bsb.toByteString().toByteArray()
+        return ByteArray(payload.size + 2). apply {
+            payload.copyInto(this)
+            putUInt16(payload.size, status.toUInt())
+        }
```

Comment:
this looks wrong, what happened to sw1 and sw2? Please don't remove them, these are field in the definition of what a Response APDU is.